test.py

```diff
@@ -499,6 +499,27 @@ def pattern(self) -> str:
         return "run"
 
 
+class ToolTestSuite(TestSuite):
+    """A collection of Python pytests that test tools, (and hence doesn't need a scylla instance)"""
```

Comment:
I understand what you mean, but note that we do have test/cql-pytest/test_tools.py, for tests that test tools but **do use** a Scylla instance because they want to run the tools on an actual Scylla. Maybe you can mention that these tests still need a **mock** Scylla server, so this test suite contains one.

It's sad we need to modify the main test.py to add "test suites" in this fashion, and add so much boilerplate code, but that comment is unrelated to your patch.

Reply:
> I understand what you mean, but note that we do have test/cql-pytest/test_tools.py, for tests that test tools but do use a Scylla instance because they want to run the tools on an actual Scylla. Maybe you can mention that these tests still need a mock Scylla server, so this test suite contains one.

This is just the generic test suite type, whether the actual suites using this type need to mock scylla or not is a different matter. They currently do but I don't think we should add that here. I will look into rephrasing this comment nevertheless.

Reply:
Comment rephrased in v4.